gazebo/rendering/LensFlare.cc

```diff
@@ -268,37 +271,27 @@ namespace gazebo
       lightPos.z = (imagePos.Z() > 1.75) ? -1 : 1;
 
       // check occlusion and set scale
-      // loop through all env cameras and find the cam that sees the light
-      // ray cast using that env camera to see if the distance to closest
-      // intersection point is less than light's world pos
       double occlusionScale = 1.0;
       if (lightPos.z >= 0.0)
       {
-        // loop through all env cameras
-        for (auto cam : ogreEnvCameras)
-        {
           // project light world point to camera clip space.
           auto viewProj = cam->getProjectionMatrix() * cam->getViewMatrix();
           auto pos = viewProj *
               Ogre::Vector4(Conversions::Convert(this->dataPtr->lightWorldPos));
           pos.x /= pos.w;
           pos.y /= pos.w;
           // check if light is visible
-          if (std::fabs(pos.x) <= 1 && std::fabs(pos.y) <= 1 && pos.z > 0)
+          if (std::fabs(pos.x) <= 1 && std::fabs(pos.y) <= 1 && pos.z > abs(pos.w))
           {
             // check occlusion using this env camera
             this->dataPtr->wideAngleDummyCamera->SetWorldPose(
-                ignition::math::Pose3d(
-                  Conversions::ConvertIgn(cam->getDerivedPosition()),
-                  Conversions::ConvertIgn(cam->getDerivedOrientation())));
-
-            occlusionScale = this->OcclusionScale(
-                this->dataPtr->wideAngleDummyCamera,
-                ignition::math::Vector3d(pos.x, pos.y, pos.z),
-                this->dataPtr->lightWorldPos);
-            break;
+            ignition::math::Pose3d(
+            Conversions::ConvertIgn(cam->getDerivedPosition()),
```

Comment:
replace `cam` with `_wideAngleCam` on this and the following line

Reply:
Same, ``_wideAngleCam`` doesn't have ``getDerivedPosition()`` here